### PR TITLE
corrected the license and updated the readme

### PR DESCRIPTION
**Pull request by KjellKod**
[README](https://github.com/KjellKod/QueueNado/blob/LogRhythm_OpenSource_Compliance/README.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,114 @@
...
+#### Test usage
+[[ShotgunAlienTests.cpp]](https://github.com/LogRhythm/QueueNado/blob/master/test/ShotgunAlienTests.cpp)
+
+
+# Headcrab - Crowbar
+`Headcrab - Crowbar` implements [request / reply](http://zguide.zeromq.org/page:all#Ask-and-Ye-Shall-Receive) messaging pattern in zmq.
+
+#### Use Cases for `Headcrab - Crowbar` 
+* Need reply to guarantee message was delivered
+* Know that their will always be a response (otherwise the socket will be broken)
```
Comment (john-gress) [inline README.md]: Rewrite suggestion: Need to receive an acknowledgement from the receiver for each message sent to it.

File: README.md [hunk 2]
```diff
@@ -1,4 +1,114 @@
...
+#### Test usage
+[[CrowbarHeadcrabTests.cpp]](https://github.com/LogRhythm/QueueNado/blob/master/test/CrowbarHeadcrabTests.cpp)
+
+
+# Harpoon - Kraken
+`Harpoon - Kraken` implements a streaming version of [pub / sub](http://zguide.zeromq.org/page:all#Getting-the-Message-Out). It enables  data streaming from a server to a client. 
+
+
+#### Kraken Purpose:
+The server sending streaming data.
```
Comment (john-gress) [inline README.md]: server -> publisher?

File: README.md [hunk 3]
```diff
@@ -1,4 +1,114 @@
...
+#### Usage example calls from the API: [[kraken.h]](https://github.com/LogRhythm/QueueNado/blob/master/src/Kraken.h)
+
+* `SendTidalWave()` : Send a data chunk to the client ([[harpoon]](https://github.com/LogRhythm/QueueNado/blob/master/src/Harpoon.h)). The call blocks until there is space available in the queue. Returns `TIMEOUT`, `INTERRUPT`, `CONTINUE` status to indicate the status of the underlying queue.
```
Comment (john-gress) [inline README.md]: client -> subscriber?

File: README.md [hunk 4]
```diff
@@ -1,4 +1,114 @@
...
+[[HarpoonBattle.h]](https://github.com/LogRhythm/QueueNado/blob/master/src/HarpoonBattle.h)
+
+#### Test usage
+* [[KrakenBattleTest.cpp]](https://github.com/LogRhythm/QueueNado/blob/master/test/KrakenBattleTest.cpp)
+* [[HarpoonKrakenTests.cpp]](https://github.com/LogRhythm/QueueNado/blob/master/test/HarpoonKrakenTests.cpp)
+* [[KrakenIntegrationTest.cpp]](https://github.com/LogRhythm/QueueNado/blob/master/test/KrakenIntegrationTest.cpp)
+
+
+# Notifier - Listener
+`Notifier - Listener` extends the `Shotgun - Alien` implementation of the  the [pub / sub](http://zguide.zeromq.org/page:all#Getting-the-Message-Out) messaging pattern in zmq.
```
Comment (john-gress) [inline README.md]: "the  the"

File: README.md [hunk 5]
```diff
@@ -1,4 +1,114 @@
...
+
+The `Notifier - Listener` classes are  wrappers around the `Shotgun - Alien`queueing framework. A `Notifier` is used to place a message onto a queue that is fed from by multiple Listeners. It uses handshake communications, where each `Listener` must respond back to the `Notifier` that it received the message. A `Notification` is deemed successful only if every expected `Listener` responds to the `Notifier`.
```
Comment (john-gress) [inline README.md]: "that is fed from" -> "that is read from"?

File: README.md [hunk 6]
```diff
@@ -1,4 +1,114 @@
...
+
+# Rifle - Vampire
+`Rifle - Vampire` implements the [push / pull](http://zguide.zeromq.org/page:all#Divide-and-Conquer) messaging pattern in zmq. The communication setup supports the following client/server communications
+
+**Client -to- Server**
+* one-to-one
+* one-to-many
+* many-to-one
+
+In a *many* scenario, it is the "one" side that has to do ```bind``` ([[Vampire.h]](https://github.com/LogRhythm/QueueNado/blob/master/src/Vampire.h) [[Rifle.h]](https://github.com/LogRhythm/QueueNado/blob/master/src/Rifle.h) on the queue.
```
Comment (KjellKod) [inline README.md]: changed.

File: README.md [hunk 7]
```diff
@@ -1,4 +1,114 @@
...
+`Shotgun - Alien` implements the [pub / sub](http://zguide.zeromq.org/page:all#Getting-the-Message-Out) messaging pattern in zmq.
+
+#### Use Cases for `Shotgun - Alien`
+* One to many: one sender communicating with many listeners.
+* Not high performance around 10k msgs a sec. This can be improved by batching many messages together.
+* Process to process communication
+
+#### Known limitations and issues
+* [Slow joiner](http://zguide.zeromq.org/php:chapter5#Representing-State-as-Key-Value-Pairs) issues don't matter or can be worked around
+* All listeners need to see every message for a topic
```
Comment (KjellKod) [inline README.md]: good point. It is rewritten and moved to use cases.

File: README.md (hunk 1 above)
Comment (KjellKod) [inline README.md]: `Need reply to guarantee message was delivered` changed to 
`Need to receive acknowledgement from the receiver for each message sent to it`

File: README.md (hunk 3 above)
Comment (KjellKod) [inline README.md]: I changed the location of `server` to be `publisher` and the Kraken text that had `client` is now referring to it as a `subscriber`

File: README.md (hunk 4 above)
Comment (KjellKod) [inline README.md]: fixed

File: README.md (hunk 5 above)
Comment (KjellKod) [inline README.md]: OK. changed.

File: README.md [hunk 8]
```diff
@@ -1,4 +1,115 @@
...
+
+# Rifle - Vampire
+`Rifle - Vampire` implements the [push / pull](http://zguide.zeromq.org/page:all#Divide-and-Conquer) messaging pattern in zmq. The communication setup supports the following client/server communications
+
+**Client -to- Server**
+* one-to-one
+* one-to-many
+* many-to-one
+
+In a *many* scenario, it is the "one" side that does the ```bind``` ([[Vampire.h]](https://github.com/LogRhythm/QueueNado/blob/master/src/Vampire.h) [[Rifle.h]](https://github.com/LogRhythm/QueueNado/blob/master/src/Rifle.h) on the queue.
```
Comment (Schatzman) [inline README.md]: Stray paren before [[Vampire.h

**Comment by Schatzman** [comment]
:+1:

**Pull request merged**